libraries/Microsoft.Bot.Builder.Azure/AzureTableStorage.cs

```diff
@@ -107,15 +107,22 @@ public AzureTableStorage(CloudStorageAccount storageAccount, string tableName)
                 return new KeyValuePair<string, object>();
             });
 
-            return (await Task.WhenAll(readTasks).ConfigureAwait(false))
-                .Where(kv => kv.Key != null);
+            var kvps = (await Task.WhenAll(readTasks).ConfigureAwait(false)).Where(kv => kv.Key != null);
+
+            IDictionary<string, object> items = new Dictionary<string, object>();
```

Comment:
Suggest you "right size" the dictionary's capacity up front based on the number of keys to avoid any growth cycles:

```CSharp
var items = new Dictionary<string, object>(keys.Length);
```

Reply:
I switched to the .ToDictionary() overload via LINQ, which makes this declaration to away. 
```
            // Wait for all the reads to complete
            var completedTasks = await Task.WhenAll(readTasks).ConfigureAwait(false);

            // Filter out the Nulls and convert to a dictionary
            var dict = completedTasks.Where(kv => kv.Key != null)
                .ToDictionary(kvp => kvp.Key, kvp => kvp.Value); 
```
If there's a way to right-size that, please let me know!